eliot/tests/test_output.py

```diff
@@ -503,6 +511,78 @@ def raiser(i):
                               (key, value) in message.items()))
 
 
+    def test_destinationExceptionCaught(self):
+        """
+        If a destination throws an exception, an appropriate error is
+        logged.
+        """
+        logger = Logger()
+        logger._destinations = Destinations()
+        dest = BadDestination()
+        logger._destinations.add(dest)
+
+        message = {"hello": 123}
+        logger.write({"hello": 123})
+        assertContainsFields(
+            self, dest[0],
+            {"message_type": "eliot:destination_failure",
+             "message": logger._safeUnicodeDictionary(message),
+             "reason": "ono",
+             "exception": "eliot.tests.test_output.MyException"})
+
+
+    def test_destinationMultipleExceptionsCaught(self):
+        """
+        If multiple destinations throw an exception, an appropriate error is
+        logged for each.
+        """
+        logger = Logger()
+        logger._destinations = Destinations()
+        logger._destinations.add(BadDestination())
+        logger._destinations.add(lambda msg: 1/0)
+        messages = []
+        logger._destinations.add(messages.append)
+
+        try:
+            1/0
+        except ZeroDivisionError as e:
+            zero_divide = str(e)
+        zero_type = ZeroDivisionError.__module__ + ".ZeroDivisionError"
+
+        message = {"hello": 123}
+        logger.write({"hello": 123})
+        self.assertEqual(
+            messages,
+            [message,
+             {"message_type": "eliot:destination_failure",
+              "message": logger._safeUnicodeDictionary(message),
+              "reason": "ono",
+              "exception": "eliot.tests.test_output.MyException"},
+             {"message_type": "eliot:destination_failure",
+              "message": logger._safeUnicodeDictionary(message),
+              "reason": zero_divide,
+              "exception": zero_type}])
+
+
+    def test_destinationExceptionCaughtTwice(self):
+        """
+        If a destination throws an exception, and the logged error about
+        it also causes an exception, then just drop that exception on the
+        floor, since there's nothing we can do with it.
+        """
+        logger = Logger()
+        logger._destinations = Destinations()
+
+        def always_raise(message):
+            raise ZeroDivisionError()
+        logger._destinations.add(always_raise)
+
+        # No exception raised; since everything is dropped no other
+        # assertions to be made.
```

Comment:
How about calling `warnings.warn`?  At least that'd give us something to test here.

It just seems a bit strange that this test has no assertions. 
Earlier tests had failing Destinations too.
Only difference here is that the only destination is failing every time.
Not sure.